File: skybay888_ui/src/services/CustomerServiceRequestService.js
import http from "../http-common"; 

class CustomerServiceRequestService {
  getAllCustomerServiceRequests(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/customerServiceRequest/customerServiceRequests`, searchDTO);
  }

  get(customerServiceRequestId) {
    return this.getRequest(`/customerServiceRequest/${customerServiceRequestId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/customerServiceRequest?field=${matchData}`, null);
  }

  addCustomerServiceRequest(data) {
    return http.post("/customerServiceRequest/addCustomerServiceRequest", data);
  }

  update(data) {
  	return http.post("/customerServiceRequest/updateCustomerServiceRequest", data);
  }
  
  uploadImage(data,customerServiceRequestId) {
  	return http.postForm("/customerServiceRequest/uploadImage/"+customerServiceRequestId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CustomerServiceRequestService();

File: src/main/java/com/skybay888/service/CustomerServiceRequestService.java
package com.skybay888.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skybay888.domain.CustomerServiceRequest;
import com.skybay888.dto.CustomerServiceRequestDTO;
import com.skybay888.dto.CustomerServiceRequestSearchDTO;
import com.skybay888.dto.CustomerServiceRequestPageDTO;
import com.skybay888.dto.CustomerServiceRequestConvertCriteriaDTO;
import com.skybay888.service.GenericService;
import com.skybay888.dto.common.RequestDTO;
import com.skybay888.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CustomerServiceRequestService extends GenericService<CustomerServiceRequest, Integer> {

	List<CustomerServiceRequest> findAll();

	ResultDTO addCustomerServiceRequest(CustomerServiceRequestDTO customerServiceRequestDTO, RequestDTO requestDTO);

	ResultDTO updateCustomerServiceRequest(CustomerServiceRequestDTO customerServiceRequestDTO, RequestDTO requestDTO);

    Page<CustomerServiceRequest> getAllCustomerServiceRequests(Pageable pageable);

    Page<CustomerServiceRequest> getAllCustomerServiceRequests(Specification<CustomerServiceRequest> spec, Pageable pageable);

	ResponseEntity<CustomerServiceRequestPageDTO> getCustomerServiceRequests(CustomerServiceRequestSearchDTO customerServiceRequestSearchDTO);
	
	List<CustomerServiceRequestDTO> convertCustomerServiceRequestsToCustomerServiceRequestDTOs(List<CustomerServiceRequest> customerServiceRequests, CustomerServiceRequestConvertCriteriaDTO convertCriteria);

	CustomerServiceRequestDTO getCustomerServiceRequestDTOById(Integer customerServiceRequestId);







}





